junkie/core_context.py

```diff
@@ -89,6 +89,9 @@ def _call(self, factory_func: Callable, stack: ExitStack, instance_name: str):
         self.logger.debug("%s = %s(%s)", instance_name, factory_func.__name__, list(args.keys()))
         instance = factory_func(**args)
 
+        if hasattr(factory_func, "__junkie_cached__"):
+            self._instances[instance_name] = instance
+
         if hasattr(instance, "__enter__"):
             self.logger.debug("%s.__enter__()", instance_name)
             stack.push(lambda *exception_details: self.logger.debug("%s.__exit__()", instance_name))
```

Comment:
As far as I remember using the ExitStack twice or more because of a reused object does not work.
So probably, we need to prevent calling `__enter__()` twice for cached objects.

Could you please add a unit test with a cached context manager object?

Reply:
So we need a test ;-)